### ISSUE TITLE
Разработка формата хранения/передачи телеметрии

### ISSUE THREAD

**Issue by glitchcore** [b4ck5p4c3/aerospace#5]
Что у нас есть:
* Напряжение питания (1 байт)
* Внешняя температура (1 байт)
* Внутренняя температура (1 байт)
* Давление (2 байта)
* Давление с доп. датчика (2 байта)
* Уровнень GSM-сигнала и статус регистрации (1 байт)
* Уровень влажности (1 байт)
* Данные акселерометра (3 байта)
* GPS по 2-м приемникам (13 * 2 байт)
    * широта (4 байта)
    * долгота (4 байта)
    * высота (2 байта)
    * скорость (1 байт)
    * направление (1 байт)
    * HDOP/VDOP, status (1 байт)

Итого 38 байт пакет. Даже в лору влезет (на самом злом SF максимальный размер 51 байт).

Можем побить передачу на части, какие-то данные передавать чаще чем другие:
* Навигация (широта, долгота, высота, скорость, направление, давление — 16 байт) передавать в каждом пакете, навигацию по-очереди с двух приемников
* Оставшиеся данные (10 байт) разбить на две части

Тогда размер пакета окажется порядка 22 байт. time on air порядка 1.5 сек, т.е. можно посылать 1 пакет в 2 секунды.

Если писать каждую секунду, то 133 КБ/час, т.е. на весь полет хватит 512 КБ

**Comment by AlexShishko** [comment]
Я бы ещё добавил что-то похожее на timestamp чтобы следить за порядком и иметь полную хронологию сообщений

**Comment by glitchcore** [comment]
Да, можно порядковый номер пакета передавать. Я не хочу прям TS пихать, у нас вроде никакого кеширования не планируется, так что момент отправки почти равен моменту приема.

**Comment by glitchcore** [comment]
Еще можно сэкономить 3*2 байта, не передавая экспоненту в координатах (приведя число к 24-битам и зафиксировав точку).
Также можно в частых пакетах отправлять только младший байт координаты.

**Comment by heeeyflo** [comment]
> Можем побить передачу на части, какие-то данные передавать чаще чем другие:

Будет круто. В голову приходят разные метрики, которые не то чтобы прям часто нужны, но которые было бы неплохо получить. Например, позиция разрыва шара (широта, долгота, высота), которая позволит на приёмной станции посчитать более точный предикшн места падения, даже если в сам момент бёрста связь отсутствовала, а пакет мы приняли уже во время падения.

> Я бы ещё добавил что-то похожее на timestamp чтобы следить за порядком и иметь полную хронологию сообщений

Да, нужен будет какой-то ключ дедупликации на случай, если будет несколько модемов или несколько приёмных станций, объединёных в одну сеть. Порядковый номер подойдёт идеально, если будет гарантироваться его уникальность и монотонное возрастание.
